dex/farm/src/lib.rs

```diff
@@ -613,4 +613,214 @@ pub trait Farm:
             &context.get_input_attributes().unwrap().compounded_reward,
         )
     }
+
+    //This function is neccessary in old farms.
+    #[payable("*")]
+    #[endpoint(migrateToNewFarm)]
+    fn migrate_to_new_farm(
+        &self,
+        #[var_args] orig_caller_opt: OptionalArg<ManagedAddress>,
+    ) -> EsdtTokenPayment<Self::Api> {
+        assert!(self, !self.farm_configuration().is_empty(), b"empty config");
+        let config = self.farm_configuration().get();
+        assert!(self, config.is_old, b"bad config");
+
+        let new_farm_address = config.new_farm_address;
+        let mut context = self.new_farm_context(OptionalArg::None);
+
+        // Same as Exit Farm, since we want to migrate the rewards and compounded tokens also
+        {
+            self.load_state(&mut context);
+            assert!(
+                self,
+                context.get_contract_state().unwrap() == &State::Migrate,
+                ERROR_NOT_MIGRATION
+            );
+
+            self.load_farm_token_id(&mut context);
+            assert!(
+                self,
+                !context.get_farm_token_id().unwrap().is_empty(),
+                ERROR_NO_FARM_TOKEN,
+            );
+
+            self.load_farming_token_id(&mut context);
```

Comment:
might need a check that farming_token_id is not empty? It is the same case as with farm_token_id

Reply:
if it is empty it will signal error so its fine without the check